[WK2] Xcode module verifier unable to find JSBase.h in some production builds

emw-apple opened the pull request:
#### cb15275e0895dce6dd7882f3914c8bc1b34c5d0f
<pre>
[WK2] Xcode module verifier unable to find JSBase.h in some production builds
<a href="https://bugs.webkit.org/show_bug.cgi?id=262573">https://bugs.webkit.org/show_bug.cgi?id=262573</a>
rdar://115357654

Reviewed by Alexey Proskuryakov.

Add include flags to OTHER_MODULE_VERIFIER_FLAGS when making STP and
downlevel builds, which references SDK headers in
$(WK_LIBRARY_HEADERS_FOLDER_PATH) and $(WK_OVERRIDE_FRAMEWORKS_DIR).
Normally, the module verifier shouldn&apos;t be given special include paths
because its purpose is to validate modules as _clients_ will import
them. But in these build styles, we control all the clients, and we are
intentionally putting header content in non-standard places to avoid
overlapping the base SDK, so the include paths are appropriate.

On Xcode 14.3 and below, these flags get passed to the tool before other
XCBuild-provided flags do, making it impossible to pass flags to the
frontend with &quot;--&quot;. As a result, disable module verification on those
versions.

The versions of XCBuild we _can_ use recognize a new build setting,
ENABLE_MODULE_VERIFIER, and have deprecated the old
CLANG_MODULES_ENABLE_VERIFIER_TOOL setting, so switching to the new
variable effectively disables it on older Xcodes.

* Source/WebKit/Configurations/WebKit.xcconfig:

Canonical link: <a href="https://commits.webkit.org/269117@main">https://commits.webkit.org/269117@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ec210eec5d18fd1202d59157d1b1776af8332aa8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/21595 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/26/builds/21903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/22643 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/23457 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/20002 "Built successfully") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/23/builds/25273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/22139 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/21182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21822 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/23/builds/25273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18710 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/24307 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/23/builds/25273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25864 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/23/builds/25273 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19769 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23717 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/20257 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/17261 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/19578 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5159 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23811 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/20164 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Source/WebKit/Configurations/WebKit.xcconfig:
```diff
@@ -31,8 +31,9 @@ PRODUCT_BUNDLE_IDENTIFIER = com.apple.$(PRODUCT_NAME:rfc1034identifier);
 DEFINES_MODULE = YES;
-CLANG_MODULES_ENABLE_VERIFIER_TOOL = $(CLANG_MODULES_ENABLE_VERIFIER_TOOL_$(USE_INTERNAL_SDK));
-CLANG_MODULES_ENABLE_VERIFIER_TOOL_YES = $(WK_NOT_$(WK_XCODE_VERSION_BEFORE_14_3_$(XCODE_VERSION_MAJOR)));
+ENABLE_MODULE_VERIFIER = $(USE_INTERNAL_SDK);
```
We no longer need to check for Xcode 14.3?

emw-apple replied:
Yeah, these build settings were renamed in Xcode 14.3, so by switching to the new names we are matching existing behavior and keeping the verifier enabled for >= 14.3 only.

aproskuryakov replied:
May be worth adding a comment, as people could be surprised to not see it run despite `ENABLE_MODULE_VERIFIER = YES`.

emw-apple committed:
[WK2] Xcode module verifier unable to find JSBase.h in some production builds
https://bugs.webkit.org/show_bug.cgi?id=262573
rdar://115357654

Reviewed by Alexey Proskuryakov.

Add include flags to OTHER_MODULE_VERIFIER_FLAGS when making STP and
downlevel builds, which references SDK headers in
$(WK_LIBRARY_HEADERS_FOLDER_PATH) and $(WK_OVERRIDE_FRAMEWORKS_DIR).
Normally, the module verifier shouldn't be given special include paths
because its purpose is to validate modules as _clients_ will import
them. But in these build styles, we control all the clients, and we are
intentionally putting header content in non-standard places to avoid
overlapping the base SDK, so the include paths are appropriate.

On Xcode 14.3 and below, these flags get passed to the tool before other
XCBuild-provided flags do, making it impossible to pass flags to the
frontend with "--". As a result, disable module verification on those
versions.

The versions of XCBuild we _can_ use recognize a new build setting,
ENABLE_MODULE_VERIFIER, and have deprecated the old
CLANG_MODULES_ENABLE_VERIFIER_TOOL setting, so switching to the new
variable effectively disables it on older Xcodes.

* Source/WebKit/Configurations/WebKit.xcconfig:

Canonical link: https://commits.webkit.org/269117@main
